src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2074,6 +2082,25 @@ namespace winrt::TerminalApp::implementation
         {
             _actionDispatch->DoAction(action);
         }
+
+        // After handling all the actions, then re-check the tabIndex. We might
+        // have been called as a part of a tab drag/drop. In that case, the
+        // tabIndex is actually relevant, and we need to move the tab we just
+        // made into position.
+        if (!firstIsSplitPane && tabIndex != -1)
+        {
+            // Move the currently active tab to the requested index Use the
```

Comment:
I don't love this. It's a bit circuitous.

Is there really no way for us to make a tab _without_ going through the positioning logic, so that we can decide where to put it later?

This feels a bit like... we have to ask it nicely to make a tab and then find out where the new tab finally landed before we do more stuff with it.

Reply:
We don't really have a "new tab at index N" action, cause we never needed one. 

We do have `_CreateNewTabFromPane` which takes an index, but `newTab` goes into `_HandleNewTab` which calls into `_OpenNewTab` which uses the default `insertPosition=-1` when it eventually calls to `_CreateNewTabFromPane`, so that we do the whole `GlobalSettings().NewTabPosition()` thing. 

And putting the `insertAtIndex` into `NewTabArgs` feels... dirty?